### PR TITLE
Changed pronouns for correctness and inclusivity

### PR DESCRIPTION
**Pull request by bikinibabe**
Changed pronouns for correctness and inclusivity. 

Hey guys, I know this is silly and pedantic, but I think this would be a fun change to make. I met a few of you at some conferences and thought this would be a fun way to practice learning git. If this works out, I might use it for a youtube tutorial on git. ;)

### CODE REVIEW

**Comment by gavinandresen** [comment]
ACK-- should be pulled into @theuni 's  trivial changes tree.

**Comment by TheBlueMatt** [comment]
Nice!

**Comment by theuni** [comment]
ACK. Pulled into trivial-next, so you can close this. In the short-term future for pulls like this, please PR them directly to the trivial-next branch in https://github.com/theuni/bitcoin . These will be merged into mainline bitcoin every few weeks.

This has been setup to reduce the PR load on the main bitcoin repo. If this goes well, a bitcoin-trivial tree will likely be setup under the bitcoin organization.

**Comment by daira** [comment]
@theuni - I don't seem to be able to file a pull request directly to theuni/bitcoin branch trivial-next (from, say, Electric-Coin-Company/bitcoin branch trivial-next): theuni/bitcoin doesn't show up in the drop-down for the target repo. At first I though this was because Electric-Coin-Company/bitcoin isn't forked from theuni/bitcoin, but it doesn't seem to be that because I can file PRs to other sibling (not parent) repos in the fork tree. Could there be some permissions problem?

(apologies for being off-topic for this PR)

**Comment by luke-jr** [comment]
NACK, this is just gramatically incorrect. "He" is gender-neutral already, so "or she" is redundant, and "they" is plural.

**Comment by daira** [comment]
This is something on which opinions differ; hopefully we can avoid rehashing a discussion that has played out on the net thousands of times before. Suffice it to say that a majority of English speakers do not find "he" to be gender-neutral and do find singular they to be perfectly grammatical. The latter has been an ordinary and widely used part of the language for hundreds of years (see http://www.crossmyt.com/hc/linghebr/austheir.html ).

**Comment by daira** [comment]
Also, I really hope we can avoid something like this: http://geekfeminism.wikia.com/wiki/Libuv_gendered_pronouns_patch_dispute

**Comment by luke-jr** [comment]
I've never seen this pushed for in reality outside of political agendas, which we should ignore IMO. Gender-neutral "he" is well-supported historically, and I do not think there is any evidence the majority of people misinterpret it as anything but gender-neutral.

**Comment by daira** [comment]
Sigh. Denying attempts to make language inclusive *is* a political agenda.

**Comment by luke-jr** [comment]
The language is already "inclusive".

**Comment by daira** [comment]
"Already" as it is on master, or as it is on trivial-next?

**Comment by luke-jr** [comment]
As it is before this PR.

**Comment by daira** [comment]
The PR has already been accepted. If you're going to revert it, that's taking a political stance. I already pointed out a previous incident of such a reversion which I think it is really highly undesirable to repeat: http://geekfeminism.wikia.com/wiki/Libuv_gendered_pronouns_patch_dispute

**Comment by luke-jr** [comment]
It has not been accepted. It should be rejected and left using correct English, rather than this change to broken English serving a political agenda. I agree we should not let this attempted political move fork the project, but that is best done by rejecting it unanimously.

**Comment by daira** [comment]
@luke-jr - *your opinion* isn't "correct English"; there is very clearly a dispute (running for hundreds of years, and so rather unlikely to be resolved in this pull request) about what "correct English" is for this case. The most straightforward, inclusive, and correct choice is that "he" refers to people who use "he" as their preferred pronoun.

Also there is no significant argument that using either "they" or "he or she" is "complicating things" relative to "he". All three are equally simple; that isn't the issue here.

**Comment by daira** [comment]
In particular, if I were an attacker trying to bypass ASLR after exploiting a vulnerability in bitcoind and you misgendered me as "he", I'd be pissed about it :-p

**Comment by luke-jr** [comment]
Trying to change English's gender-neutral "he" into something else is sexist and many people find it objectionable. Please keep your political agenda out of Bitcoin code. It's really that simple.

Also, this is now totally tangent to the PR, but I don't care if you get pissed at proper English; that is your problem, not mine. I will continue to use generic "he" pronouns for everyone online, as I do not know or care what sex they are.

**Comment by gavinandresen** [comment]
Closed; will be merged the next time the trivial changes tree is merged.

Luke: stop it.

Daira: please ignore Luke, he has a history of being a "poisonous person."

**Comment by luke-jr** [comment]
What's next? Are we going to add anarchist leanings into the code too? (at least the anarchists have contributed something..)

Frankly, it's disappointing anyone has ACK'd this nonsense. And now @gavinandresen is resorting to personal attacks based on a 3 year old grudge. Sigh.

**Comment by sipa** [comment]
Funny how nobody seems to actually have looked at the changes. Node0 is not
a person, and shouldn't have been 'he' in the first place; 'it' is
perfectly fine.

Alao, @gmaxwell says that part of software development is offending as few
people as possible, so if we have language that offends anarchists, we
should probably change it too, so long as we can preserve accuracy. It's
like writing portable software.

**Comment by sipa** [comment]
In fact, the last change refers to a master process, and isn't about a
person either. It too can just be 'it'.

**Comment by ripper234** [comment]
Please see https://github.com/bikinibabe/bitcoin/pull/1

'he or she' doesn't cover all genders. 'They' should be preferred as a more gender-neutral term.

File: qa/rpc-tests/wallet.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def run_test (self):
         walletinfo = self.nodes[0].getwalletinfo()
         assert_equal(walletinfo['immature_balance'], 0)
 
-        # Have node0 mine a block, thus he will collect his own fee. 
+        # Have node0 mine a block, thus they will collect their own fee.
```
Comment (laanwj) [inline qa/rpc-tests/wallet.py]: Should be 'it'. A node is one entitiy not a multiple.

File: src/checkqueue.h [hunk 2]
```diff
@@ -81,7 +81,7 @@ class CCheckQueue
                     fAllOk &= fOk;
                     nTodo -= nNow;
                     if (nTodo == 0 && !fMaster)
-                        // We processed the last element; inform the master he can exit and return the result
+                        // We processed the last element; inform the master he or she can exit and return the result
```
Comment (laanwj) [inline src/checkqueue.h]: The master is not a person in this case, and again should be 'it'. Let's not antromorphize data structures.

**Comment by laanwj** [comment]
I agree that attackers should not be a 'he', and that change is good, but I disagree how this was ACKed and merged seemingly without thinking about better alternatives for the other two changes. Software objects are not gendered.

**Comment by daira** [comment]
I wasn't able to file a pull request to theuni/bitcoin, but see https://github.com/Electric-Coin-Company/bitcoin/compare/trivial-next...refine-pronouns-and-aslr for a change to use singular "they" for people and "it" for software entities. I also added a doc improvement about the effect of ASLR. This should be preferred to bikinibabe#1 which introduces a grammatical mistake (but thanks for the intention).

@gavinandresen - message received and understood :-)

**Comment by luke-jr** [comment]
"They" is gramatically plural, so this this change could instead read:
```
Attackers who are able to cause execution of code at an arbitrary memory location may be thwarted if they don't know where anything useful is located.
```

**Comment by laanwj** [comment]
@diara ACK on https://github.com/Electric-Coin-Company/bitcoin/compare/trivial-next...refine-pronouns-and-aslr

**Comment by daira** [comment]
@luke-jr That changes the meaning in a way that does not reflect the intent. Each attacker potentially has different knowledge about "where anything useful is located". Any particular attacker trying to bypass ASLR faces a problem whose difficulty is dependent on that attacker's knowledge, and not on other attackers' knowledge. The statement is not dependent on no attackers knowing "where anything useful is located", and so it is imprecise to lump all attackers together, or all attackers' knowledge together. I realise that a reader could nevertheless interpret the plural statement as intended, but it can also be misinterpreted in a way that the singular "An attacker..." statement can't.

**Comment by luke-jr** [comment]
Maybe this?

Attackers who are able to cause execution of code at an arbitrary memory location may be thwarted if they don't *personally* know where anything useful is located.

**Comment by daira** [comment]
An individual attacker can also be a group of people with shared knowledge. "They" is the correct pronoun to use to cover all possible cases (a person, a group, an autonomous bot, etc.)

I'm going to stop commenting now because my patch at https://github.com/Electric-Coin-Company/bitcoin/compare/trivial-next...refine-pronouns-and-aslr has already been ACKed, and there seems to be no opposition besides @luke-jr .

**Comment by TheBlueMatt** [comment]
...and this is why open source lacks diversity of viewpoints, and suffers for it :(

**Comment by daira** [comment]
@TheBlueMatt - yes. As a trans person, it can be exhausting. Simple improvements shouldn't encounter this degree of resistance (and to be fair, often don't, but it's the times when they do that leave the lasting impression).

**Pull request closed without merging**